addons/sale/models/sale_order.py

```diff
@@ -630,19 +634,25 @@ def _onchange_pricelist_id_show_update_prices(self):
 
     #=== CRUD METHODS ===#
 
+    def _get_so_name(self, code, seq_date):
+        self.ensure_one()
+        return self.env['ir.sequence'].next_by_code(code, sequence_date=seq_date) or self._default_name
+
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:
             if 'company_id' in vals:
                 self = self.with_company(vals['company_id'])
-            if vals.get('name', _('New')) == _('New'):
+        orders = super().create(vals_list)
+        default_name = self._default_name()
+        for order in orders:
+            # Override the name only if it has the default value.
+            if order.name == default_name:
                 seq_date = fields.Datetime.context_timestamp(
-                    self, fields.Datetime.to_datetime(vals['date_order'])
-                ) if 'date_order' in vals else None
-                vals['name'] = self.env['ir.sequence'].next_by_code(
-                    'sale.order', sequence_date=seq_date) or _('New')
-
-        return super().create(vals_list)
+                    self, fields.Datetime.to_datetime(order.date_order)
+                ) if order.date_order else None
+                order.name = order._get_so_name('sale.order', seq_date)
+        return orders
```

Comment:
Can't you find a way to keep the sequence computation pre-create to insert directly the correct value ?

Reply:
The base subscription sequence (SUB00XXX) is chosen based on the computed is_subscription field. If this key is not in the values, the name will be wrong. I thought it was easier to set the name on an existing record. Maybe it could be fixed before the call to super with a precompute=True on the is_subscription field definition? The is_subscription value would be available ?

Reply:
If you pop the 'name' from the create values when it's the default value, and specify name & is_subscription as precomputed fields, it should work :thinking: (but it's a bit hacky)

Reply:
is_subscription depends on the sale_temporal type of the order_lines but this one is not stored, so precompute has no effect. :-/

Reply:
Wait, is_subscription is stored but not its dependencies? it's strange :thinking:

Reply:
The sol pricing is stored, product_template_id is a related on the line and it is providing line.product_template_id.recurring_invoice. When the pricing is set and recurring_invoice is true, the temporal_type is 'subscription'. `is_subscription` depends on temporal_type because it was shorter but I can make it depends on order_line.pricing_id and order_line.product_template_id.recurring_invoice if you prefer. But still, it is the temporal type that should be evaluated because it depends on both conditions. We can store the temporal_type but I thought it was not necessary/overkill.